analyses/cnv-comparison/util/cnv-comparison-functions.R

```diff
@@ -0,0 +1,169 @@
+
+  # Rearrange the columns
+  cnv_df <- cnv_df %>%
+    dplyr::select("chrom", "loc.start", "loc.end", "ID", "num.mark", "seg.mean") %>%
+    dplyr::mutate(chrom = factor(chrom, levels = paste0("chr", c(1:22, "X", "Y"))))
+
+  return(cnv_df)
+
+}
+
+filter_segmean <- function(cnv_df, segmean_cutoff){
+  # Given the data.frame containing CNV caller output, return a data.frame with
+  # a column with the values 0 and 1 to represent loss and gain, respectively.
+  #
+  # Args:
+  #   cnv_df: data.frame containing CNV caller output
+  #
+  # Return:
+  #   filtered_cnv_df: the data.frame given, returned with a column containing
+  #                    an aberration label for loss or gain
+
+  # Rename columns for GenVisR function downstream
+  colnames(cnv_df) <-
+    c("chromosome",
+      "start",
+      "end",
+      "sample",
+      "probes",
+      "segmean")
+
+  filtered_cnv_df <- cnv_df %>%
+    dplyr::mutate(aberration = dplyr::case_when(segmean < -segmean_cutoff ~ 0,
+                                  segmean > segmean_cutoff ~ 1)) %>%
+    dplyr::filter(!is.na(aberration))
+
+  return(filtered_cnv_df)
+
+}
+
+plot_violin <- function(filtered_cnv_df) {
+  # Given the data.frame filtered for size of aberrations, plot the proportion
+  # or frequency (denoted by the plot_type argument) of aberrations across
+  # chromosomes with `geom_violin`
+  #
+  # Args:
+  #   filtered_cnv_df: data.frame filtered for cutoff size of aberrations
+  #   plot_title: a title string for the plot produced
+  #
+  # Return:
+  #   violin_plot: violin plot depicting the log2 transformed sizes of
+  #                aberrations detected across chromosomes
+
+  # Create violin plot where the y-axis represents the log2 transformed segmean
+  violin_plot <- ggplot2::ggplot(filtered_cnv_df,
+                             ggplot2::aes(x = chromosome,
+                                          y = (log2(segmean) + 1))) +
+    ggplot2::geom_violin() +
+    ggplot2::theme_bw() +
+    ggplot2::ylab("Log transformed segmean values")
+
+  return(violin_plot)
+
+}
+
+plot_histology_barplot <- function(filtered_cnv_df, metadata) {
+  # Given the data.frame filtered by cutoff segmean value with `filter_segmean`,
+  # plot the proportion of aberrations across chromosomes with `geom_barplot`
+  #
+  # Args:
+  #   filtered_cnv_df: data.frame filtered for cutoff size of aberrations and
+  #                    joined with the metadata
+  #   metadata: the relevant metadata data.frame
+  #
+  # Return:
+  #   barplot: barplot depicting the proportion of aberrations detected across
+  #            chromosomes, annotated by the `broad_histology`` variable in the
+  #            metadata
+
+  # Create a data.frame with the filtered dataframe joined with the metadata
+  meta_joined <- filtered_cnv_df %>%
+    dplyr::inner_join(metadata, by = c("sample" = "Kids_First_Biospecimen_ID"))
+
+  # Create barplot where the y-axis represents the size of aberration
```

Comment:
On second look, I am pretty confused by these plots. What is actually being plotted here? I had thought it was the number of each type in each chromosome, but the `stat = "identity"` is confusing me. Would the default `stat = "count"` make more sense? 

On some investigation, it looks like these ultimately produce the same plots as the ones that I was expecting, but if you leave off the `y` aesthetic, and use the default `geom_bar()` (no stat) you get a much smaller file size and better y axis labels.